### PR TITLE
add participants validation

### PR DESCRIPTION
**Pull request by andibraeu**
validates participants markdown files against a json schema. maybe we need to discuss that schema.

replaces #402

### CODE REVIEW

**Comment by andibraeu** [comment]
Unfortunately PR comments for this validation don't work for PRs created from forks, as all permissions are read-only then

File: participants-schema.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "mastodon": {
+      "type": "string",
+      "pattern": "^(http|https)://[A-z0-9.\\-/]+/@[A-z0-9._%+\\-]+$"
+    },
+    "image": {
+      "$ref": "#/definitions/url"
+    },
+    "contact-via": {
+      "type": "string",
+      "pattern": "^(^[A-z0-9äöüÄÖUß_\\-\\.\\+]+@[A-z0-9äöüÄÖUß_\\-\\.]+\\.[A-Za-z]{2,}$|[\\+]?[(]?[0-9]{3}[)]?[-\\s\\.]?[0-9]{3}[-\\s\\.]?[0-9]{4,6})$"
```
Comment (coderbyheart) [inline participants-schema.json]:
```suggestion
```
There is no specific format for this field.

File: participants-schema.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Participants Schema",
+  "type": "object",
+  "definitions": {
+    "url": {
+      "type": "string",
+      "pattern": "^(http|https)://[A-Za-z0-9ÄÖÜäöüß\\-_\\./]+"
```
Comment (coderbyheart) [inline participants-schema.json]:
```suggestion
      "pattern": "^https?://\S+$"
```

Don't forget all the other alphabets.

File: participants-schema.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+  },
+  "properties": {
+    "name": {
+      "type": "string"
```
Comment (coderbyheart) [inline participants-schema.json]:
```suggestion
      "type": "string",
      "minLength": 2
```

File: participants-schema.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "pronouns": {
+      "type": "string"
```
Comment (coderbyheart) [inline participants-schema.json]:
```suggestion
      "type": "string",
      "minLength": 2
```

Comment (coderbyheart) [reply participants-schema.json]: Otherwise empty string works.

File: participants-schema.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "link": {
+      "$ref": "#/definitions/url"
+    },
```
Comment (coderbyheart) [inline participants-schema.json:22]: This should be merged with "url" and only link should be allowed.

File: participants-schema.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    "mastodon": {
+      "type": "string",
+      "pattern": "^(http|https)://[A-z0-9.\\-/]+/@[A-z0-9._%+\\-]+$"
```
Comment (coderbyheart) [inline participants-schema.json]:
```suggestion
      "pattern": "^https?://[A-z0-9.\\-/]+/@\S+$"
```

File: participants-schema.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "image": {
+      "$ref": "#/definitions/url"
+    },
+    "contact-via": {
+      "type": "string",
+      "pattern": "^(^[A-z0-9äöüÄÖUß_\\-\\.\\+]+@[A-z0-9äöüÄÖUß_\\-\\.]+\\.[A-Za-z]{2,}$|[\\+]?[(]?[0-9]{3}[)]?[-\\s\\.]?[0-9]{3}[-\\s\\.]?[0-9]{4,6})$"
+    },
+    "room-type": {
+      "type": "string"
```
Comment (coderbyheart) [inline participants-schema.json]:
```suggestion
      "type": "string",
      "minLength": 2
```

File: participants-schema.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "linkedin": {
+      "type": "string",
+      "pattern": "^(http|https)://([a-z]{2,3}\\.|)linkedin\\.com/in/[A-z0-9\\-]+"
+    },
+    "matrix": {
+      "type": "string",
+      "pattern": "@[A-z0-9._=-]+:[A-z0-9.-]+\\.[A-z]{2,}"
```
Comment (coderbyheart) [inline participants-schema.json]:
```suggestion
      "pattern": "@\S+:\S+"
```

Again, a bit strict on the domain names.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: participants-schema.json (hunk 5 above)
Comment (andibraeu) [reply participants-schema.json:22]: image is an url, too, that's why I extracted this type to a definition

Comment (andibraeu) [reply participants-schema.json:20]: image is an url, too, that's why I extracted that type to a definition

File: participants-schema.json (hunk 1 above)
Comment (andibraeu) [reply participants-schema.json]: In template it's defines as "email or phone number", but I don't have a strong optinion. We maybe should add a minLength?

File: participants-schema.json (hunk 2 above)
Comment (andibraeu) [reply participants-schema.json]: I'm not sure, if that works, as regex in JSON Schema are limited. At least, we need to escape the backslash

Comment (andibraeu) [reply participants-schema.json]: it works :)

File: participants-schema.json (hunk 1 above)
Comment (andibraeu) [reply participants-schema.json]: I removed patternd and added a minLength condition

**Comment by coderbyheart** [comment]
Thanks!

**Pull request closed without merging**